optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -231,7 +254,15 @@ SolverConfigBuildStep recordAndRegisterBeans(OptaPlannerRecorder recorder, Recor
 
         additionalBeans.produce(new AdditionalBeanBuildItem(OptaPlannerBeanProvider.class));
         unremovableBeans.produce(UnremovableBeanBuildItem.beanTypes(OptaPlannerRuntimeConfig.class));
-        return new SolverConfigBuildStep(solverConfig);
+        return new SolverConfigBuildItem(solverConfig);
+    }
+
+    private void disableANC(SolverConfig solverConfig) {
+        if (solverConfig.getScoreDirectorFactoryConfig().getDroolsAlphaNetworkCompilationEnabled() != null
+                && solverConfig.getScoreDirectorFactoryConfig().getDroolsAlphaNetworkCompilationEnabled()) {
```

Comment:
Shouldn't it be enough to check for `== true`?

Reply:
NPE if getDroolsAlphaNetworkCompilationEnabled is null (it Boolean, not boolean)

Reply:
Interesting, I honestly thought that `null == true` is `false`, not an NPE. Well, I learn every day.

Reply:
It might be; I know `if(Boolean)` can lead to a NPE if the boolean is null, but that might be handled differently then `if(Boolean == true)`

Reply:
It the same for the `Boolean == true` case (auto unboxing): https://stackoverflow.com/a/25824276